orttraining/orttraining/core/optimizer/concat_training_rewriter.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/optimizer/rewrite_rule.h"
+
+namespace onnxruntime {
+
+/**
+@Class ConcatTrainingRewriter
+
+Rewrite rule that replaces Concat with ConcatTraining, that has an additional output 
+used in building the gradient for Concat node.
+
+It is attempted to be triggered only on nodes with op type "Concat".
+*/
+class ConcatTrainingRewriter : public RewriteRule {
+ public:
+  ConcatTrainingRewriter() noexcept : RewriteRule("ConcatTrainingRewriter") {}
```

Comment:
>ConcatTrainingRewriter [](start = 51, length = 22)

"Rewrite" is somewhat generic. how about `ConcatTrainingReplacement`?

Reply:
Changing to "ConcatReplacement" since "ConcatTrainingReplacement" is somewhat misleading